flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1278,7 +1278,12 @@ public void run() {
     }
 
     void scheduleTriggerRequest() {
-        timer.execute(this::executeQueuedRequest);
+        if (isShutdown()) {
```

Comment:
This is only an optimization because `timer` thread can be shutdown after the check.
I'm concerned that adding `synchronized` might create deadlocks (if not now then in the future).
See also thread [above](https://github.com/apache/flink/pull/14683#discussion_r559742642).